### ISSUE TITLE
[BUG] After Upgrade K8s, I have much errors: Failed to rotate log for container...

### ISSUE THREAD

**Issue by DaniDD** [rancher/rancher#40811]
**Rancher Server Setup**
- Rancher version: v2.7.1
- Installation option (Docker install/Helm Chart):
  in Docker Container
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.24.10-rancher4-1
- Cluster Type (Local/Downstream):
  vSphere, generated by Rancher v2.4.x

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
Administrator

**Describe the bug**
Any days ago I have upgrade my K8s Cluster from v1.23.xx to v 1.24.9. On the same day I have renew any nodes with Ubuntu 20.04 and Docker 20.10 
From this time Errors are flooding my kubelet logs.
`{"log":"E0308 06:08:49.427364 1136819 container_log_manager.go:266] \"Failed to rotate log for container\" err=\"failed to rotate log \\\"/var/log/pods/kube-system_snapshot-controller-6984fdc566-wpzxk_dd9e9890-c44b-47c7-9adf-9577acba9056/snapshot-controller/0.log\\\": failed to reopen container log \\\"ab0a9bf26504ee3dcb0784a6a75370a68a83fb1beff820c0ca1e94813d18f712\\\": rpc error: code = Unknown desc = docker does not support reopening container log files\" path=\"/var/log/pods/kube-system_snapshot-controller-6984fdc566-wpzxk_dd9e9890-c44b-47c7-9adf-9577acba9056/snapshot-controller/0.log\" containerID=\"ab0a9bf26504ee3dcb0784a6a75370a68a83fb1beff820c0ca1e94813d18f712\"\n","stream":"stderr","time":"2023-03-08T06:08:49.427448137Z"}
{"log":"E0308 06:08:49.431059 1136819 remote_runtime.go:1145] \"ReopenContainerLog from runtime service failed\" err=\"rpc error: code = Unknown desc = docker does not support reopening container log files\" containerID=\"343a81ff1805ee7148ef342941093d98b786691023bd47af455f541ed41bdcbf\"\n","stream":"stderr","time":"2023-03-08T06:08:49.43116302Z"}
{"log":"E0308 06:08:49.431145 1136819 container_log_manager.go:266] \"Failed to rotate log for container\" err=\"failed to rotate log \\\"/var/log/pods/kasten-io_executor-svc-c4d559bcb-vlph6_111453b3-a17d-4cca-85c1-a458cb3c6d87/executor-svc/0.log\\\": failed to reopen container log \\\"343a81ff1805ee7148ef342941093d98b786691023bd47af455f541ed41bdcbf\\\": rpc error: code = Unknown desc = docker does not support reopening container log files\" path=\"/var/log/pods/kasten-io_executor-svc-c4d559bcb-vlph6_111453b3-a17d-4cca-85c1-a458cb3c6d87/executor-svc/0.log\" containerID=\"343a81ff1805ee7148ef342941093d98b786691023bd47af455f541ed41bdcbf\"\n","stream":"stderr","time":"2023-03-08T06:08:49.431196458Z"}
{"log":"E0308 06:08:49.435576 1136819 remote_runtime.go:1145] \"ReopenContainerLog from runtime service failed\" err=\"rpc error: code = Unknown desc = docker does not support reopening container log files\" containerID=\"b112b70bc26802b791232299547a55b66dc2d38e0e79d4c00c253ad345702a47\"\n","stream":"stderr","time":"2023-03-08T06:08:49.435691377Z"}
{"log":"E0308 06:08:49.435755 1136819 container_log_manager.go:266] \"Failed to rotate log for container\" err=\"failed to rotate log \\\"/var/log/pods/cattle-monitoring-system_pushprox-kube-controller-manager-proxy-58f5d844c6-8zsmj_94f96b7f-1877-47ac-9e21-3de449753e3e/pushprox-proxy/0.log\\\": failed to reopen container log \\\"b112b70bc26802b791232299547a55b66dc2d38e0e79d4c00c253ad345702a47\\\": rpc error: code = Unknown desc = docker does not support reopening container log files\" path=\"/var/log/pods/cattle-monitoring-system_pushprox-kube-controller-manager-proxy-58f5d844c6-8zsmj_94f96b7f-1877-47ac-9e21-3de449753e3e/pushprox-proxy/0.log\" containerID=\"b112b70bc26802b791232299547a55b66dc2d38e0e79d4c00c253ad345702a47\"\n","stream":"stderr","time":"2023-03-08T06:08:49.435816912Z"}
{"log":"E0308 06:08:49.440137 1136819 remote_runtime.go:1145] \"ReopenContainerLog from runtime service failed\" err=\"rpc error: code = Unknown desc = docker does not support reopening container log files\" containerID=\"706c616c7c83cdbb07e45b71869722a425dff614ee23f0f165dcb617021e547f\"\n","stream":"stderr","time":"2023-03-08T06:08:49.440407292Z"}`

**Expected Result**
less errors in my logs

**Comment by YohannHammad** [comment]
Seems to be related to [#39819](https://github.com/rancher/rancher/issues/39819)

**Comment by vmule** [comment]
I've seen the same issue:

`Jul 28 06:09:55 cane kubelet[319]: E0728 06:09:55.895505     319 remote_runtime.go:736] "ReopenContainerLog from runtime service failed" err="rpc error: code = Unknown desc = docker does not support reopening container log files" containerID="aed4876bb8b3366c15fb3bfdf3d01f78853ad6971157e0013fc0c4e19efafefc"
Jul 28 06:09:55 cane kubelet[319]: E0728 06:09:55.895756     319 container_log_manager.go:268] "Failed to rotate log for container" err="failed to rotate log \"/var/log/pods/kube-app-1_kube-app-1-prod-86c86674dd-r7hh5_9437804d-f918-4dcd-aca8-1e20a67f88f1/nginx/1.log\": failed to reopen container log \"aed4876bb8b3366c15fb3bfdf3d01f78853ad6971157e0013fc0c4e19efafefc\": rpc error: code = Unknown desc = docker does not support reopening container log files" path="/var/log/pods/kube-app-1_kube-app-1-prod-86c86674dd-r7hh5_9437804d-f918-4dcd-aca8-1e20a67f88f1/nginx/1.log" containerID="aed4876bb8b3366c15fb3bfdf3d01f78853ad6971157e0013fc0c4e19efafefc"
Jul 28 06:10:06 cane kubelet[319]: E0728 06:10:06.104602     319 remote_runtime.go:736] "ReopenContainerLog from runtime service failed" err="rpc error: code = Unknown desc = docker does not support reopening container log files" containerID="aed4876bb8b3366c15fb3bfdf3d01f78853ad6971157e0013fc0c4e19efafefc"`

**Comment by ZleFox** [comment]
I am having the same log rotate error on a rke1 cluster on ubuntu 22.04 with docker 23.0.5 
is there a "fix" ?

**Comment by YohannHammad** [comment]
Yes there is a « fix » : https://github.com/rancher/rancher/issues/39819#issuecomment-1472278470

**Comment by ZleFox** [comment]
creating a daemon.json with ` {  "log-driver": "json-file",
    "log-opts": { "max-size": "10m", "max-file": "3" }
}` max-age and restarting the daemon fixed it

**Comment by DaniDD** [comment]
I have now Rancher v2.7.6 und K8s v1.26.8 and have no more the error

**Comment by techviewleo** [comment]
> creating a daemon.json with ` { "log-driver": "json-file", "log-opts": { "max-size": "10m", "max-file": "3" } }` max-age and restarting the daemon fixed it
This worked for me too

**Comment by DiegoSanchezPiedra** [comment]
> creating a daemon.json with  { "log-driver": "json-file", "log-opts": { "max-size": "10m", "max-file": "3" } } max-age and restarting the daemon fixed it

This got rid of the erros messages for me too, thanks!
